app/grandchallenge/components/forms.py

```diff
@@ -79,3 +84,166 @@ def save(self, *args, **kwargs):
 
     class Meta:
         fields = ("user_upload", "creator", "comment")
+
+
+class MultipleCIVCreateForm(Form):
+    _possible_widgets = {
+        *InterfaceFormField._possible_widgets,
+    }
+
+    def __init__(self, *args, instance, base_obj, user, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.instance = instance
+        self.user = user
+        self.base_obj = base_obj
+
+        for slug, values in base_obj.values_for_interfaces.items():
+            current_value = None
+
+            if instance:
+                current_value = instance.values.filter(
+                    interface__slug=slug
+                ).first()
+
+            interface = ComponentInterface.objects.get(slug=slug)
+
+            if interface.is_image_kind:
+                self.fields[slug] = self._get_image_field(
+                    interface=interface,
+                    values=values,
+                    current_value=current_value,
+                )
+            elif interface.requires_file:
+                self.fields[slug] = self._get_file_field(
+                    interface=interface,
+                    values=values,
+                    current_value=current_value,
+                )
+            else:
+                self.fields[slug] = self._get_default_field(
+                    interface=interface, current_value=current_value
+                )
+
+    def _get_image_field(self, *, interface, values, current_value):
+        return self._get_default_field(
+            interface=interface, current_value=current_value
+        )
+
+    def _get_file_field(self, *, interface, values, current_value):
+        return self._get_default_field(
+            interface=interface, current_value=current_value
+        )
+
+    def _get_default_field(self, *, interface, current_value):
+        return InterfaceFormField(
+            instance=interface,
+            initial=current_value.value if current_value else None,
+            required=False,
+            user=self.user,
+        ).field
+
+    def clean(self):
+        cleaned_data = super().clean()
+        try:
+            cleaned_data.update(self._process_new_interfaces())
+        except ValidationError as e:
+            self.errors.update(e.message_dict)
```

Comment:
We don't need to raise something here?

Reply:
Changed implementation, ValidationError is now raised in `_process_new_interfaces`.